akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+Some projects may prefer to store events in human-readable formats instead of formats optimised for write/read throughput.
+The style of serialization used in Akka Persistence is completely left up to the concrete Journal plugin implementations -
+this means that some community provided journals may offer human readable serialization, however it is not the default style
+to persist events.
+
+Additionally some datastores are able to store "documents" as JSON (or BSON) natively, wich may have additional benefits
+for later debugging or searchability through the events. In general we highly encourage the use projections using
+:ref:`akka-persistence-query-scala` to migrate data stored using a binary format (optimised for write throughput)
+stored on the "write side" of the application, to the "read side" of the application – which more often than not
+is a completely different database, which for example is more suited for storing the events in a human readable format.
+
+One of the possible ways of supporting this is to emit "the type that the journal understands how to store natively"
+(e.g. a JSON type), in the ``EventAdapter`` such that the Journal can directly store it in the datastore without applying
+any additional serialization. This technique is described as sub-section of the :ref:`detach-domain-from-data-model-scala` pattern.
+
+.. note::
+  If in need of human-readable events on the *write-side* of youe application reconsider if preparing materialized views
+  using :ref:`akka-persistence-query-scala` would not be an efficient way to go about this, without compromising the
+  write-side's throughput characteristics.
+
+  If indeed you want to use a human-readable representation on the write-side, pick a Persistence plugin
+  that provides that functionality, or – implenent one yourself.
+
+
+Schema evolution in action
+==========================
+
+In this section we will discuss various schema evolution techniques using concrete examples and explaining
+some of the various options one might go about handling the described situation. The list below is by no means
+a complete guide, so feel free to adapt these techniques depending on your serializer's capabilities
+and/or other domain specific limitations.
+
+Add fields
+----------
+
+**Situation:**
+You need to add a field to an existing message type. For example, a ``SeatReservation(code: String)`` now
+needs to have an associated code which indicates if it is a window or aisle seat.
+
+**Solution:**
+Adding fields is the most common change you'll need to apply to your messages so make sure the serialization format
+you picked for your payloads can handle it apropriately, i.e. such changes should be *binary compatible*.
+This is easily achieved using the right serializer toolkit – we recommend something like `Google Protocol Buffers`_ or
+`Apache Thrift`_ however other tools may fit your needs just as well – picking a serializer backend is something
```

Comment:
I'm not sure yet about adding too many links in the docs here.
We could add an "add your helpful libs list", like this:

```

Helpful serialization libraries
-------------------------------
This section aims to provide a list of projects which may be helpful to build your serializers with,
by for example providing scala bindings to otherwise very "java style" serialization libraries.

Serialization:
- `ScalaPB`_ - is protobuf compiler plugin that generates Scala sources instead of Java sources like the plain protoc compiler does.

Various:
- `Stamina`_ - is a project aimed to help with schema evolution where the schema for each event is versioned (V1, V2, ...).
```

However I'm slightly worried that it may get outdated soon.